lib/parsers/zip_parser/file_reader.rb

```diff
@@ -18,6 +18,7 @@ def message
       'Could not find the EOCD signature in the buffer - maybe a malformed ZIP file'
     end
   end
+  InvalidCentralDirectory = Class.new(Error)
```

Comment:
Wouldn't it be better to inherit from `StandardError`?

Reply:
It inherits from `FileReader::Error`, which in turn inherits from `StandardError`. `FileReader::Error` is rescued in the parser modules and returns a `nil`, so that's deliberate.

Reply:
Oops, my bad then. For some reason I thought it's the top level `Exception` class in Ruby 😅